### DISCUSSION TITLE
FLAX always offline

### DISCUSSION THREAD

**Discussion by steamch** [guydavis/machinaris#346] [Q&A] [answered]
System setup:
OS: UNRAID 6.9.2
Docker version number (output of docker info): 20.10.5
Machinaris branch: Main
Machinaris version number: 0.6.0

FLAX Farming: Offline but the Chia shows Farming: Active
also i found when i click on the Connections,the web will shows 
`DataTables warning: table id=conns_Tower_chia - Cannot reinitialise DataTable. For more information about this error, please see http://datatables.net/tn/3`

**Comment by guydavis** [comment, marked as answer]
Hi! Sorry to hear you encountered this issue of duplicate/stale worker status on upgrade.  Please follow the [upgrade guide](https://github.com/guydavis/machinaris/wiki/Unraid#how-do-i-update-from-v05x-to-v060-with-fork-support) first.

Then try this:

1. Stop all Machinaris containers in the Unraid UI |  Docker tab.
2. Remove file /mnt/user/appdata/machinaris/machinaris/dbs/machinaris.db holding stale status records.
3. Check for updates for containers.
4. Start up all Machinaris containers
4. Wait about 10-15 minute for all the containers to re-populate active status.

Ideally, the stale records are gone and only active status remains. Please let me know how it goes.  If this solves your issue, please mark the Discussion as Answered.

**Reply by steamch** [reply]
`Internal Server Error
The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.`

after i followed your step.i got a new problem when i click on the Connections,the web will shows that

**Reply by guydavis** [reply]
Please attach /mnt/user/appdata/machinaris/machinaris/logs/webui.log

**Reply by steamch** [reply]
[https://github.com/steamch/personal-RP/blob/main/webui.log](url)
thanks for your help

**Reply by guydavis** [reply]
> [https://github.com/steamch/personal-RP/blob/main/webui.log](https://github.com/steamch/personal-RP/blob/main/webui.log) thanks for your help

Thanks for the log.  Here's the relevant section with error when you browse to the Connections page. I don't see errors on other pages.
```
TypeError: exceptions must derive from BaseException
2021-10-22 10:50:48 [1135] [ERROR] Exception on /connections [GET]
Traceback (most recent call last):
  File "/chia-blockchain/venv/lib/python3.9/site-packages/flask/app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "/chia-blockchain/venv/lib/python3.9/site-packages/flask/app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/chia-blockchain/venv/lib/python3.9/site-packages/flask/app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "/chia-blockchain/venv/lib/python3.9/site-packages/flask/app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/machinaris/web/routes.py", line 214, in connections
    connections = chia.load_connections_show()
  File "/machinaris/web/actions/chia.py", line 129, in load_connections_show
    return Connections(connections)
  File "/machinaris/web/models/chia.py", line 234, in __init__
    'farmer_port': self.blockchain_port(connection.blockchain),
  File "/machinaris/web/models/chia.py", line 264, in blockchain_port
    raise("Unknown blockchain fork of selected: " + blockchain)
TypeError: exceptions must derive from BaseException
```
Are you trying to also farm Flora?  If so, that will be supported in v0.6.1, not in v0.6.0.  I would recommend you change the Repository line of your Machinaris (main) container to read:

```
ghcr.io/guydavis/machinaris:test
```
like this:
![image](https://user-images.githubusercontent.com/12936/138526580-797d684c-ec8c-4615-8b44-f6bce13607cd.png)

Test is the early version of upcoming v0.6.1.  Details about Machinaris [Release Streams](https://github.com/guydavis/machinaris/wiki/Releases#release-streams)  Hope this helps!  Feel free to let me know if you have further questions.

**Reply by steamch** [reply]
oh,it works.thank you
i also want to know how to import key in other blockchains? like flax hddcoin.
the dashboard's wallets only shows chia

**Reply by guydavis** [reply]
Hi, please see https://github.com/guydavis/machinaris/wiki/Keys#copy-to-forks

**Reply by steamch** [reply]
yes,
but now i get a another problem
like in the Summary the flax shows 'Farming: Active',in the Blockchains the flax shows 'Full Node Synced'.
but in the Wallets the flax shows 'Sync status: Not synced',and in the alert flax report Your harvester appears to be offline
i am so confused about it
thanks for you answer

**Reply by steamch** [reply]
my bad.it's my internet problem
but when i update to 0.6.1
the chia wallet has been disappeared
[https://imgur.com/g796ahZ](https://imgur.com/g796ahZ)

**Reply by guydavis** [reply]
Hi, what do you see when you get into the container with `docker exec -it machinaris bash`  and type: `chia wallet show` Do you get an error?    If so, please try restarting the Machinaris container (or just the Chia services with `chia start farmer -r`)  Also, please provide ~/.machinaris/machinaris/logs/apisrv.log as there may be some clues there.  Thanks!

**Reply by steamch** [reply]
[https://raw.githubusercontent.com/steamch/personal-RP/main/apisrv.log](https://raw.githubusercontent.com/steamch/personal-RP/main/apisrv.log)
when i restart the chia wallet is still missing-------------now is working
thanks for you helping
